### ISSUE TITLE
Request for Position: Canvas Formatted Text

### ISSUE THREAD

**Issue by littledan** [WebKit/standards-positions#122]
## Request for position on an emerging web specification

* WebKittens who can provide input: @litherum 

### Information about the specification

* Title: Canvas Formatted Text
* GitHub repository: https://github.com/WICG/canvas-formatted-text/
* Explainer:
    * https://github.com/WICG/canvas-formatted-text/blob/main/README.md
    * https://github.com/WICG/canvas-formatted-text/blob/main/explainer-datamodel.md
    * https://github.com/WICG/canvas-formatted-text/blob/main/explainer-metrics.md
    * https://github.com/WICG/canvas-formatted-text/blob/main/explainer-rendering.md
* Authors: @dlibby-, @ijprest, @sushraja-msft, @travisleithead

### Anything else we need to know

I previously worked on Intl.Segmenter, which was attempting to solve parts of this problem (linguistically aware line breaking, a frequent request which we have heard in ECMA-402) in a way @litherum explained turns out to be simply incorrect, https://github.com/tc39/proposal-intl-segmenter-v2/issues/8.

An alternative for some use cases would be to avoid using canvas for text and use HTML rendering. This is possible for some applications and not others.

The reason I'm asking about this API: Bloomberg is looking at improvements to text rendering in the charts within one of our iOS apps, which are rendered in a `<canvas>` in a WKWebView, with text currently drawn with the fillText API. We're looking at this API in parallel with investigating whether we could take the HTML approach to avoid its need.

**Comment by fantasai** [comment]
It appears this has been superseded by [HTML-in-Canvas](https://github.com/WICG/html-in-canvas). Should we close this issue?

**Comment by petamoriken** [comment]
Based on [the minutes from the Web Engine Hackerfest](https://github.com/Igalia/webengineshackfest/wiki/2026-Extended-Canvas-Text-Metrics-%E2%80%90-moving-it-forward), it seems there is a discussion that HTML-in-Canvas does not fully supersede it, and Canvas Formatted Text is still necessary.

Here is the new explainer: https://github.com/Igalia/explainers/tree/main/canvas-formatted-text
